client/packages/programs/src/JsonForms/components/ProgramEvent.tsx

```diff
@@ -0,0 +1,228 @@
+              .rest(z.string().optional())
+          ),
+        }),
+      ])
+      .optional(),
+  })
+  .strict();
+type Options = z.infer<typeof Options>;
+
+const extractAt = (datetime?: string, options?: Options): Date => {
+  if (!datetime) {
+    return new Date();
+  }
+  const beforeDate = new Date(new Date(datetime).getTime() - 1);
+  if (!options || !options.at) {
+    return beforeDate;
+  }
+
+  if (options.at.type === 'encounter') {
+    switch (options.at.encounterStartDatetime) {
+      case 'before':
+        return beforeDate;
+      case 'after':
+        return new Date(datetime);
+      default:
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        ((_: never) => {})(options.at.encounterStartDatetime as never);
+    }
+  } else if (options.at.type === 'programEnrolment') {
+    switch (options.at.programEnrolmentDatetime) {
+      case 'before':
+        return beforeDate;
+      case 'after':
+        return new Date(datetime);
+      default:
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        ((_: never) => {})(options.at.programEnrolmentDatetime as never);
+    }
+  }
+
+  return new Date();
+};
+
+const getDisplayOptions = (
+  data: string | null | undefined,
+  options?: Options
+) => {
+  const show =
+    options?.display?.type === 'string' ? options?.display?.show : null;
+
+  if (!show) {
+    return data;
+  }
+
+  const displayValue = show.find(value => value[0] === data)?.[1];
+  return displayValue ?? '';
+};
+
+const UIComponent = (props: ControlProps) => {
+  const { label, uischema, config } = props;
+  const [datetime, setDatetime] = React.useState<string | undefined>();
+  const patientId = config?.patientId;
+  const encounterDocName = config?.documentName.includes('Encounter');
+  const enrolmentDocName = config?.documentName.includes('Program');
+
+  const { data: currentEncounter } = useEncounter.document.byDocName(
+    encounterDocName ? config.documentName : undefined
+  );
+  const { data: programEnrolment } = useProgramEnrolments.document.byDocName(
+    enrolmentDocName ? config.documentName : undefined
+  );
+
+  const { errors, options } = useZodOptionsValidation(
+    Options,
+    uischema.options
+  );
+
+  useEffect(() => {
+    if (
+      options?.at?.type === 'encounter' &&
+      options?.at?.encounterStartDatetime
+    ) {
+      setDatetime(currentEncounter?.startDatetime);
+    }
+
+    if (
+      options?.at?.type === 'programEnrolment' &&
+      options?.at?.programEnrolmentDatetime
+    ) {
+      setDatetime(programEnrolment?.enrolmentDatetime);
+    }
+  }, [currentEncounter, programEnrolment]);
+
+  const { data: events } = useProgramEvents.document.list({
+    at: options?.at ? extractAt(datetime, options) : undefined,
+    patientId: patientId ?? '',
+    filter: {
+      type: {
+        equalTo: options?.eventType,
+      },
+      documentType: {
+        equalTo: options?.documentType,
+      },
```

Comment:
```suggestion
      },
      documentName:
        options?.documentType === undefined
          ? {
              equalTo: config.documentName,
            }
          : undefined,
```

I think this is missing, e.g. if documentType is not specify use the events for the current document...

Reply:
🤔 I made documentType a required option....

Reply:
ok I see, we also need to be able to display events for the current document though, e.g. `VL Pending` / `VL Result` which is for a particular encounter.

```
{
  /** Show a program events that target documents of a certain type.
Note, this only includes events that are not targeting a specific document name.
If not specified the program events for the current document are displayed (using the document name of the current JSONForms document)  */
  documentType?: string,
}
```